src/Cli/dotnet/commands/dotnet-workload/install/WorkloadGarbageCollector.cs

```diff
@@ -73,7 +73,14 @@ void GarbageCollectWorkloadSets()
             var resolver = GetResolver();
 
             var installedWorkloadSets = resolver.GetWorkloadManifestProvider().GetAvailableWorkloadSets();
-            
+
+            // This version corresponds to the version currently in use, whether from a global.json, install state, etc.
+            var versionInUse = resolver.GetWorkloadVersion();
+            if (installedWorkloadSets.ContainsKey(versionInUse))
+            {
+                WorkloadSetsToKeep.Add(versionInUse);
+            }
```

Comment:
Sounds good. I was very iffy about this bit because on the one hand, what I have here may GC a workload set you need for another repo, and I don't think we have a precise way to know that without basically scanning your whole drive for global.json files and combining them all. That's bad. On the other hand, your solution means we never properly GC workload sets, which means that if someone uses them for a long time and updates regularly, they'll take up an unbounded space and eventually overflow your computer. I don't like either solution, but I don't have a better option in mind...

I'll leave the install state logic for now; it should be relevant after this change, but it won't hurt, and we may want to move back to it at some point.

Reply:
When we do garbage collection for workload sets, we will keep track of what global.json files we've used, so we will be able to find them without scanning the whole disk.

Reply:
This is the method that does garbage collection for workload sets, and it identified the workload set specified in the global.json file in the current directory as one we didn't need and scheduled it for deletion...